sdk/go/common/util/gitutil/git.go

```diff
@@ -848,3 +848,80 @@ func (r byShortNameLengthDesc) Swap(i, j int) { r[i], r[j] = r[j], r[i] }
 func (r byShortNameLengthDesc) Less(i, j int) bool {
 	return len(r[j].Short()) < len(r[i].Short())
 }
+
+type detectGitAccessors interface {
+	Stat(name string) (detectGitFileInfo, error)
+}
+type detectGitFileInfo interface {
+	IsDir() bool
+}
+
+type detectGitAccessorsImpl struct{}
+
+func (a *detectGitAccessorsImpl) Stat(name string) (detectGitFileInfo, error) {
+	return os.Stat(name)
+}
+
+// As go-git do not support an analogous to `git rev-parse --show-toplevel`,
+// I am bringing this from a suggestion in an open issue tracking the requirement
+// https://github.com/go-git/go-git/issues/74#issuecomment-647779420
+func DetectGitRootDirectory(path string) (string, error) {
+	return detectGitRootDirectory(path, &detectGitAccessorsImpl{})
+}
+
+func detectGitRootDirectory(path string, accessors detectGitAccessors) (string, error) {
```

Comment:
mmm I am probably missing something as I am not seeing any repo method to get the relative path from the current working directory to the root of the git repo.

Lets say the git repo root is `/home/user/repo` and the working dir is `/home/user/repo/src/something`, I dont see a way to get `src/something` . My first thought was to use something like `git rev-parse --show-toplevel` but git-go does not have support for such thing.

Is there a way using PlainOpen to get what is the relative path to the repo root by imputing `/home/user/repo/src/something`?

By running PlainOpen in child directory (ie `/home/user/repo/src/something` instead of the root, I get repository does not exist

Reply:
Discussed on slack, I was not aware of the `DetectDotGit` 

```
	repo, err = git.PlainOpenWithOptions("./node_modules", &git.PlainOpenOptions{DetectDotGit: true})
	if err != nil {
		panic(err)
	}
	worktree, err = repo.Worktree()
	if err != nil {
		panic(err)
	}
	root = worktree.Filesystem.Root()
	fmt.Println(root)
```

it works beautifully, I will use this instead

Reply:
done, applied in https://github.com/pulumi/pulumi/pull/16585